File: examples/csharp/EventsPublish.cs
// KubeMQ MCP Example: events_publish
//
// Demonstrates the events_publish tool using the official MCP C# SDK.
// Publishes an ephemeral event (fire-and-forget).
//
// JSON-RPC Request:
// {
//   "jsonrpc": "2.0",
//   "id": 5,
//   "method": "tools/call",
//   "params": {
//     "name": "events_publish",
//     "arguments": {
//       "channel": "example-events",
//       "body": "Event data",
//       "metadata": "event-meta",
//       "tags": {"source": "mcp-example"}
//     }
//   }
// }
//
// JSON-RPC Response:
// {
//   "jsonrpc": "2.0",
//   "id": 5,
//   "result": {
//     "content": [{ "type": "text", "text": "Event published successfully to channel 'example-events'" }],
//     "isError": false
//   }
// }

using ModelContextProtocol.Client;

class EventsPublish
{
    static async Task Main(string[] args)
    {
        var url = Environment.GetEnvironmentVariable("KUBEMQ_MCP_URL") ?? "http://localhost:9090";
        var transport = new HttpClientTransport(new HttpClientTransportOptions { Endpoint = new Uri($"{url}/mcp") });
        await using var client = await McpClientFactory.CreateAsync(transport);

        var result = await client.CallToolAsync("events_publish", new Dictionary<string, object>
        {
            ["channel"] = "example-events",
            ["body"] = "Event data",
            ["metadata"] = "event-meta",
            ["tags"] = new Dictionary<string, string> { ["source"] = "mcp-example" },
        });

        Console.WriteLine($"Tool: events_publish");
        Console.WriteLine($"Result: {result}");
    }
}

// Expected output:
// Tool: events_publish
// Result: Event published successfully to channel 'example-events'

File: examples/kotlin/EventsPublish.kt
// KubeMQ MCP Example: events_publish
//
// Demonstrates the events_publish tool using the MCP Kotlin SDK.
// Publishes an ephemeral event — lost if no subscriber is active.
//
// JSON-RPC Request:
// {
//   "jsonrpc": "2.0",
//   "id": 5,
//   "method": "tools/call",
//   "params": {
//     "name": "events_publish",
//     "arguments": {
//       "channel": "example-events",
//       "body": "Event data"
//     }
//   }
// }
//
// JSON-RPC Response:
// {
//   "jsonrpc": "2.0",
//   "id": 5,
//   "result": {
//     "content": [{"type": "text", "text": "Event published successfully to channel 'example-events'"}],
//     "isError": false
//   }
// }

import io.modelcontextprotocol.kotlin.sdk.Implementation
import io.modelcontextprotocol.kotlin.sdk.client.Client
import io.modelcontextprotocol.kotlin.sdk.client.StreamableHttpClientTransport
import io.ktor.client.*
import io.ktor.client.plugins.sse.*
import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    val url = System.getenv("KUBEMQ_MCP_URL") ?: "http://localhost:9090"

    val httpClient = HttpClient { install(SSE) }
    val transport = StreamableHttpClientTransport(client = httpClient, url = "$url/mcp")
    val client = Client(clientInfo = Implementation(name = "kubemq-mcp-kotlin-example", version = "1.0.0"))
    client.connect(transport)

    val result = client.callTool("events_publish", mapOf(
        "channel" to "example-events",
        "body" to "Event data"
    ))

    println("Tool: events_publish")
    println("Result: $result")

    client.close()
    httpClient.close()
}

// Expected output:
// Tool: events_publish
// Result: Event published successfully to channel 'example-events'
